test/plugin.test.ts

```diff
@@ -102,6 +119,77 @@ describe("scalaJSPlugin", () => {
       .toBeNull();
   }, testOptions);
 
+  testBothModes(it, "works with a project with subprojects", async (mode, suffix) => {
+    const [plugin, fakePluginContext] = setup({
+      subprojects: [
+        {
+          projectID: "otherProject",
+          uriPrefix: "foo",
+        },
+        {
+          projectID: null,
+          uriPrefix: "bar",
+        },
+      ]
+    });
+
+    await plugin.configResolved.call(undefined, { mode: mode });
+    await plugin.buildStart.call(fakePluginContext, {});
+
+    expect(normalizeSlashes(await plugin.resolveId.call(fakePluginContext, 'foo:main.js')))
+      .toContain('/testproject/other-project/target/scala-3.2.2/otherproject-' + suffix + '/main.js');
+    expect(normalizeSlashes(await plugin.resolveId.call(fakePluginContext, 'bar:main.js')))
+      .toContain('/testproject/target/scala-3.2.2/testproject-' + suffix + '/main.js');
+
+    expect(await plugin.resolveId.call(fakePluginContext, 'scalajs/main.js'))
+      .toBeNull();
+  }, testOptions);
+
+  it.fails("with duplicate projectID", async () => {
+    setup({
+      subprojects: [
+        {
+          projectID: "otherProject",
+          uriPrefix: "foo",
+        },
+        {
+          projectID: "otherProject",
+          uriPrefix: "bar",
+        },
+      ]
+    });
+  });
+
+  it.fails("with duplicate uriPrefix", async () => {
+    setup({
+      subprojects: [
+        {
+          projectID: "otherProject",
+          uriPrefix: "foo",
+        },
+        {
+          projectID: null,
+          uriPrefix: "foo",
+        },
+      ]
+    });
+  });
+
+  it.fails("when both projectID and subproojects are specified", async () => {
+    setup({
+      projectID: "xxx",
+      subprojects: []
+    });
+  });
+
+  it.fails("when both uriPrefix and subproojects are specified", async () => {
+    setup({
+      uriPrefix: "xxx",
+      subprojects: []
```

Comment:
Is `subprojects: []` something that should be supported at all? Probably not.

Reply:
Yeah, it is a kind of an edge case.
* Theoretically, one might use it dynamically and no subprojects would equal to disabled plugin.
* Practically, such use case is unlikely to be common.
* It also might require the plugin to handle those edge cases. (Well, I am not sure if printSbtTasks handles empty list well…)

So, I'll disable this edge case.